library/Booster/Pattern/ApplyEquations.hs

```diff
@@ -537,102 +497,100 @@ evaluatePattern' Pattern{term, constraints, ceilConditions} = do
   No iteration happens at the same AST level inside these traversals,
   one equation will be applied per level (if any).
 -}
-applyTerm ::
-    MonadLoggerIO io =>
-    Direction ->
-    EquationPreference ->
-    Term ->
-    EquationT io Term
-applyTerm direction pref trm = do
-    logOtherNS "booster" (LevelOther "Simplify") "Calling equation-based simplifier"
-    config <- getConfig -- avoid re-reading config at every node
-    descend config trm
-  where
-    -- descend :: EquationConfig -> Term -> EquationT io Term
-    descend config t@(Term attributes _)
-        | attributes.isEvaluated = pure t
-        | otherwise =
-            fromCache Equations t >>= \case
-                Nothing -> do
-                    simplified <- apply config t
-                    toCache Equations t simplified
-                    pure simplified
-                Just cached -> do
-                    when (t /= cached) $ do
-                        setChanged
-                        emitEquationTrace t Nothing (Just "Cache") Nothing $ Success cached
-                    pure cached
-
-    -- Bottom-up evaluation traverses AST nodes in post-order but finds work top-down
-    -- Top-down evaluation traverses AST nodes in pre-order
-    apply config = \case
+traverseTerm ::
```

Comment:
I suggest we expand the documentation of this function to reference it's two current instantiations, `llvmSimplify` and `iterateEquations`.